test/integration/bulk-load-test.js

```diff
@@ -248,3 +251,91 @@ exports['bulkLoad - verify null value'] = function(test) {
   });
   connection.execSqlBatch(request);
 };
+
+exports.testStreamingBulkLoad = function(test) {
+  const totalRows = 500000;
+  const connection = this.connection;
+  const tableName = '#streamingBulkLoadTest';
+
+  connection.on('error', function(err) {
+    test.ifError(err);
+  });
+  startCreateTable();
+
+  function startCreateTable() {
+    const sql = 'create table ' + tableName + ' (i int not null primary key)';
+    const request = new Request(sql, completeCreateTable);
+    connection.execSqlBatch(request);
+  }
+
+  function completeCreateTable(err) {
+    test.ifError(err);
+    startBulkLoad();
+  }
+
+  function startBulkLoad() {
+    const bulkLoad = connection.newBulkLoad(tableName, completeBulkLoad);
+    bulkLoad.addColumn('i', TYPES.Int, {nullable: false});
+    const rowStream = bulkLoad.getRowStream();
+    connection.execBulkLoad(bulkLoad);
+    const rowSource = new RowSource(totalRows);
+    rowSource.pipe(rowStream);
+  }
+
+  function completeBulkLoad(err, rowCount) {
+    test.ifError(err);
+    test.equal(rowCount, totalRows);
+    startVerifyTableContent();
+  }
+
+  function startVerifyTableContent() {
+    const sql =
+      'select count(*) ' +
+        'from ' + tableName + ' a ' +
+          'inner join ' + tableName + ' b on a.i = b.i - 1';
+    const request = new Request(sql, completeVerifyTableContent);
+    request.on('data', (row) => {
+      test.equals(row[0].value, totalRows - 1);
+    });
+    connection.execSqlBatch(request);
+  }
+
+  function completeVerifyTableContent(err, rowCount) {
+    test.ifError(err);
+    test.equal(rowCount, 1);
+    test.done();
+  }
+
+  class RowSource extends Readable {
+    constructor(totalRows) {
+      super({
+        objectMode: true
+      });
+      this.totalRows = totalRows;
+      this.rowCount = 0;
+    }
+    _read() {
+      while (this.rowCount < this.totalRows) {
+        const i = this.rowCount++;
+        const row = [i];
+        if (!this.push(row)) {
+          return;
+        }
+      }
+      this.push(null);
+    }
+  }
+};
+
+exports.dummyToSwitchToEncryptedConnection = function(test) {
+  // (Please let me know if you find a better solution for this.)
+  useEncryptedConnection = true;
```

Comment:
I'm wondering why this line would be necessary? 🤔

Reply:
Oh, I see, you wanted to execute one time with an encrypted connection, and one time without. I don't think that is strictly necessary, as the encryption happens on a totally different level in the network stack and I can't imagine any reason why the encryption would have any effect on the Bulkload functionality.

Reply:
In message-io.cs, flow-control takes a different path when the connection is encrypted (drain event of securePair.cleartext).